.github/workflows/tests.yml

```diff
@@ -23,84 +24,52 @@ jobs:
         with:
           PATTERNS: |
             **/**.go
+            "!test/"
             go.mod
             go.sum
-      - name: install
-        run: make install install_abci
-        if: "env.GIT_DIFF != ''"
-      - uses: actions/cache@v2.1.7
-        with:
-          path: ~/go/pkg/mod
-          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
-          restore-keys: |
-            ${{ runner.os }}-go-
+            Makefile
+      - name: Run Go Tests
+        run: |
+          make test-group-${{ matrix.part }} NUM_SPLIT=4
         if: env.GIT_DIFF
-      # Cache binaries for use by other jobs
-      - uses: actions/cache@v2.1.7
```

Comment:
this helps in speeding up subsequent runs of the same pr. Are you fine with longer build times?

Reply:
Well, this is complicated and the diff makes this harder to read.

The thing that this is caching is the binary of the ABCI test binary, which is only useful if the files it depends on don't change. Building this binary takes <1s, so we're maximally saving a second, at the cost of network traffic (putting/getting the object from the cache) and the scheduling overhead for the CI. Even if we cache objects, we still have to do things around using the cache (most notably cloning the git repository) that take a lot more time. In effect we'll end up saving time, by cloning the repository fewer times. 

These kinds of cache workflows really only make sense when you have an expensive build process (that takes minutes) and those build artifacts can be used to run many tests in parallel, with fewer dependencies (e.g. on the source tree). For these artifacts because there's no fan-out, and the tests depend on having the source tree so we're only really saving time in no-op builds, in all other cases, I'd expect us to save time. 

Even if it did make build times a bit longer, I think it'd be worth it for the simplicity gains, if only because the total wall-clock time is dominated by the build/runtime of the e2e tests (in PR/commit) which means that unless we can make that shorter there'll never be any changes to actual latency of a build, because that will always be the longest poll.